test/controllers/patients_controller_test.rb

```diff
@@ -12,6 +15,65 @@ class PatientsControllerTest < ActionDispatch::IntegrationTest
     @clinic = create :clinic
   end
 
+  describe 'index method' do
+    before do
+      # Not sure if there is a better way to do this, but just calling
+      # `sign_in` a second time doesn't work as expected
+      delete destroy_user_session_path
+    end
+
+    it 'should reject users without access' do
+      sign_in @user
+
+      get patients_path
+      assert_redirected_to root_path
+
+      get patients_path + '.csv'
```

Comment:
Thanks! I was trying to find something like this but couldn't and gave up.  The pointer is much appreciated.